app/controllers/notes_controller.rb

```diff
@@ -1,12 +1,31 @@
 class NotesController < ApplicationController
+  before_action :find_note, only: %i[edit update]
+  before_action :find_volunteer
+
   def create
-    volunteer = Volunteer.find(params[:volunteer_id])
-    volunteer.notes.create(note_params)
-    redirect_to edit_volunteer_path(volunteer)
+    @volunteer.notes.create(note_params)
+    redirect_to edit_volunteer_path(@volunteer)
+  end
+
+  def edit
+  end
```

Comment:
I think it's not necessary, rails will search for the view based on the controller's name